internal/cli/atlas/privateendpoints/azure/watch.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type WatchOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	id    string
+	store store.PrivateEndpointDescriber
+}
+
+func (opts *WatchOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *WatchOpts) watcher() (bool, error) {
+	result, err := opts.store.PrivateEndpoint(opts.ConfigProjectID(), provider, opts.id)
+	if err != nil {
+		return false, err
+	}
+	return result.Status == "WAITING_FOR_USER" || result.Status == "FAILED", nil
```

Comment:
[q] I'm not that familiar with this API but I know for private link AWS and azure have different status, flagging this just in case